tests/core/utils.py

```diff
@@ -1,11 +1,167 @@
 # First Party
+# Standard Library
+import glob
+import json
+import logging
+import os
+import shutil
+from os.path import join
+
 from smdebug.core.collection import Collection
 from smdebug.core.collection_manager import CollectionManager
-from smdebug.core.config_constants import DEFAULT_COLLECTIONS_FILE_NAME
+from smdebug.core.config_constants import (
+    DEFAULT_COLLECTIONS_FILE_NAME,
+    DEFAULT_SAGEMAKER_METRICS_PATH,
+)
+from smdebug.core.modes import ModeKeys
+from smdebug.core.reader import FileReader
+from smdebug.core.sagemaker_utils import is_sagemaker_job
+from smdebug.trials import create_trial
 
 
 def write_dummy_collection_file(trial):
     cm = CollectionManager()
     cm.create_collection("default")
     cm.add(Collection(trial))
     cm.export(trial, DEFAULT_COLLECTIONS_FILE_NAME)
+
+
+def delete_local_trials(local_trials):
+    for trial in local_trials:
+        shutil.rmtree(trial)
+
+
+# need this to seek to the right file offset for test output verification
+metrics_file_position = 0
+
+
+def check_metrics_file(save_steps, saved_scalars=None):
+    """
+    Check the SageMaker metrics file to ensure that all the scalars saved using
+    save_scalar(sm_metrics=True) or mentioned through SM_METRICS collections, have been saved.
+    """
+    global metrics_file_position
+    if is_sagemaker_job():
+        METRICS_DIR = os.environ.get(DEFAULT_SAGEMAKER_METRICS_PATH)
+        if not METRICS_DIR:
+            logging.warning("SageMaker Metric Directory not specified")
+            return
+        file_name = "{}/{}.json".format(METRICS_DIR, str(os.getpid()))
+        scalarnames = set()
+
+        import collections
+
+        train_metric = collections.defaultdict(list)
+        eval_metric = collections.defaultdict(list)
+
+        with open(file_name) as fp:
```

Comment:
can we break into smaller functions?

Reply:
We can address the refactoring and renaming of functions in different PR for code cleanup.
These particular functions are utility functions in test code and currently only getting used in save_scalar related test only.
If we encounter more usage for these functions, we will definitly reorganize them.